### DISCUSSION TITLE
Wanted - Updates to the Ruby Starter Project

### DISCUSSION THREAD

**Discussion by aurorawalker** [BattlesnakeOfficial/feedback#116] [Feature Requests]
The [Ruby Starter Project](https://github.com/BattlesnakeOfficial/starter-snake-ruby) was rotated into community support in 2021, and there are a number of possible improvements that would increase it's quality and make it more similar to the [Official Starter Projects](https://docs.battlesnake.com/references/starter-projects). 

1) Update the Battlesnake 'random' movement algorithm such that it never selects the direction that would cause it to collide with its own heck.

2) Update Comments/inline documentation:
The function-level comments to link to the Game Board data (example: https://github.com/BattlesnakeOfficial/starter-snake-python/blob/main/server_logic.py#L38)
The in-line comments to provide hints/suggestions on what next steps should be for the user. It must be similar to comments documented here: https://github.com/BattlesnakeOfficial/starter-snake-python/blob/main/server_logic.py#L63 

3) Update/create a simple test suite
Ideally the test suite is a starting place, for developers who are using the Starter Project template to build a Battlesnake, to be able to test their own code via common Rust testing tools, using built-in libraries if and when possible. The ReadMe file should also be updated to include any instructions on how to run the test suite.

It should be similar in purpose/ability as the test suites in the other official starter projects such as:
[Official Typescript Starter Project](https://github.com/BattlesnakeOfficial/starter-snake-typescript/tree/main#running-tests)
[Official Javascript Starter Project](https://github.com/BattlesnakeOfficial/starter-snake-javascript/tree/main#running-tests)
[Official Go Starter Project](https://github.com/BattlesnakeOfficial/starter-snake-go/tree/main#running-tests)
[Official Python Starter Project](https://github.com/BattlesnakeOfficial/starter-snake-python/tree/main#running-tests)

**Comment by aurorawalker** [comment]
Note - if you are looking at this because of [Hacktoberfest](https://hacktoberfest.digitalocean.com/), your pull request can be just one of these three improvements, as every bit helps!

**Comment by ChaelCodes** [comment]
Is there any way to move the Ruby starter template from master to main?

**Reply by aurorawalker** [reply]
Absolutely. We did that with other repos, but looks like this one got missed in the shuffle. On it.
